File: js/alphabet-position.js
/**
 * Tanggal: 9/3/25
 * Link: https://www.codewars.com/kata/546f922b54af40e1e90001da/train/javascript
 */

function alphabetPosition(text) {
    const alphabets = "abcdefghijklmnopqrstuvwxyz";

    return text
        .toLowerCase()
        .split("")
        .map((char) => alphabets.indexOf(char) + 1)
        .filter((index) => index !== 0)
        .join(" ");
}

console.log(alphabetPosition("abc"));
console.log(alphabetPosition("The sunset sets at twelve o' clock."));
console.log(alphabetPosition("The narwhal bacons at midnight."));

File: java/AlphabetPosition.java
/**
 * Tanggal: 1 Januari 2026
 * Link: https://www.codewars.com/kata/546f922b54af40e1e90001da/train/java
 */

public class AlphabetPosition {
    public static void main(String[] args) {
        System.out.println(alphabetPosition("Abc"));
        System.out.println(alphabetPosition("Xyz"));
        System.out.println(alphabetPosition("The sunset sets at twelve o'clock"));
    }

    static String alphabetPosition(String text) {
        String textLowerCase = text.toLowerCase();
        String result = "";

        for (int i = 0; i < textLowerCase.length(); i++) {
            char character = textLowerCase.charAt(i);
            int position = character - 96;

            if (position > 0 && position < 27) {
                result += position + " ";
            }
        }

        return result.trim();
    }
}
